sentinelhub/aws.py

```diff
@@ -483,8 +484,8 @@ class will try to find the index automatically. If there will be multiple choice
         :type config: SHConfig or None
         """
         self.tile_name = self.parse_tile_name(tile_name)
-        self.datetime = self.parse_datetime(time)
-        self.date = self.datetime.split('T')[0]
+        self.timestamp = parse_time(time, ignoretz=True)
+        self.date_str = self.timestamp.isoformat().split('T')[0]
```

Comment:
The problem is that the input parameter `time` can be either just a date or date + time. So then `self.timestamp` can be either `datetime.date` or `datetime.datetime`. On the other hand, `AwsProduct` can only obtain date information (without making additional requests). Because `AwsProduct` and `AwsTile` share some common methods then we have to define a date for both of them.

I agree this is a bit convoluted but for now, I would leave both a timestamp and a date object. I will however try storing date as a `datetime.date` object instead of a string.